### PR TITLE
Kotlin 2.0!!

### PR DESCRIPTION
**Pull request by Frank1234**
CHANGE many dependency updates and kotlin 2.0

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,30 +1,29 @@
 [versions]
-androidxCore = "1.12.0"
-androidxLifecycle = "2.7.0"
+androidxCore = "1.13.1"
+androidxLifecycle = "2.8.0"
```
Comment (sebaslogen) [inline gradle/libs.versions.toml:3]: Nope: keep it at 2.7.0 for now https://www.reddit.com/r/androiddev/comments/1csjwne/lifecycle_280_only_compatible_with_compose_17/

**Review by sebaslogen — CHANGES_REQUESTED** [review]

File: build.dep.compose.gradle [hunk 2]
```diff
@@ -17,4 +14,4 @@ dependencies {
     implementation(libs.composeUIToolingPreview)
     implementation(libs.composeMaterial3)
     implementation(libs.composeLifecycle)
-}
\ No newline at end of file
+}
```
Comment (ninovanhooff) [inline build.dep.compose.gradle:17]: ❤️ ;-)

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,30 +1,29 @@
...
 kotlinxCoroutinesTest = "1.7.3"
```
Comment (izadiegizabal) [inline gradle/libs.versions.toml:4]: i think we can also update this to `1.8.1`

https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-coroutines-test

File: gradle/libs.versions.toml (hunk 1 above)
Comment (Frank1234) [reply gradle/libs.versions.toml:3]: Ha, yes, I was pragmatic: it gave an error I didn't know how to fix, so I reverted.

Comment (sebaslogen) [reply gradle/libs.versions.toml:3]: You can now use 2.8.1 hotfix https://developer.android.com/jetpack/androidx/releases/lifecycle#2.8.1

Comment (sebaslogen) [reply gradle/libs.versions.toml]: Lifecycle is now on 2.8.2 with a supposedly final fix for the bug introduced in 2.8.0, better update before merging

**Review by sebaslogen — CHANGES_REQUESTED** [review]
Extra lib update

**Review by ninovanhooff — DISMISSED** [review]

**Review by sebaslogen — APPROVED** [review]

**Pull request merged**